android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -182,14 +188,80 @@ protected void dispatchOnDisappear() {
         .getEventDispatcher()
         .dispatchEvent(new ScreenDisappearEvent(mScreenView.getId()));
 
+    dispatchEventInChildContainers(ScreenLifecycleEvent.Disappear);
+
+    dispatchTransitionProgress(1.0f, true, true);
+    mAboveScreen = null;
+  }
+
+  public void incrementCoalescingKey() {
+    mCoalescingKey++;
+  }
+
+  protected void dispatchTransitionProgress(
+      float alpha, boolean closing, boolean shouldIncrementCoalescingKey) {
+    boolean goingForward = false;
+    if (getScreen().getContainer() instanceof ScreenStack) {
+      goingForward = ((ScreenStack) getScreen().getContainer()).isGoingForward();
+    }
+    if (shouldIncrementCoalescingKey) {
+      incrementCoalescingKey();
+    }
+    sendTransitionProgressEvent(alpha, closing, goingForward);
+
+    if (mAboveScreen != null
+        && mAboveScreen.getFragment() != null
+        && !mAboveScreen.getFragment().isSendingProgress()) {
+      if (shouldIncrementCoalescingKey) {
+        mAboveScreen.getFragment().incrementCoalescingKey();
+      }
+      // if we are in transparentModal presentation, only one screen is sending progress
+      mAboveScreen.getFragment().sendTransitionProgressEvent(alpha, !closing, goingForward);
```

Comment:
I am not sure what more than the comment in `ScreenStack.java` (https://github.com/software-mansion/react-native-screens/pull/890/files#diff-b1439496d79d456c74fa09061a5e3fef9e2e7c35e7c3ba434343ada55695abf6R259). As for removing `mAboveScreen`, we could just not send progress to non-animating screens, or maybe deal with it from JS. I am open to other options since this one is certainly not perfect.